### ISSUE TITLE
Bug: Misleading error when including non standard file types with projectService

### ISSUE THREAD

**Issue by geirmarius** [typescript-eslint/typescript-eslint#10057]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```TypeScript
// Example file, but any file with a non standard extension
// /src/App.vue

<script lang="ts" setup>
 const hello = "world"
</script>

<template>
  <h1>Hello {{hello}}</h1>
</template>
```

### ESLint Config

```javascript
//@ts-check

import ts from 'typescript-eslint';
import js from '@eslint/js';
import vue from 'eslint-plugin-vue';

export default ts.config(
  js.configs.recommended,
  ...ts.configs.strictTypeChecked,
  ...ts.configs.stylisticTypeChecked,
  ...vue.configs['flat/recommended'],

  {
    files: ['**/*.ts', '**/*.js', '**/*.vue'],
    languageOptions: {
      parserOptions: {
        parser: ts.parser,
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
      },
    },
  },

  {
    files: ['**/*.js'],
    ...ts.configs.disableTypeChecked,
  }
);
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
  "include": [ "src" ],
}
```

### Expected Result

To report an error about missing file extension in `extraFileExtensions`

### Actual Result

Reporting error about missing include in tsconfig.json:

```plaintext
/home/projects/vitejs-vite-fqss2d/src/App.vue
  0:0  error  Parsing error: /home/projects/vitejs-vite-fqss2d/src/App.vue was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject
```

### Additional Info

I've created a reproduction here: https://stackblitz.com/edit/vitejs-vite-fqss2d?file=eslint.config.mjs (it's a basic typescript project, and doesn't actually run any vue, but for linting that shouldn't be a problem)

1. let it install, then exit the server in the terminal and run npm run lint instead
2. in the terminal you can see the error when it tries to lint App.vue

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@eslint/js-eslint`     | `9.11.1` |
| `typescript-eslint`     | `8.7.0` |
| `TypeScript`                           | `5.6.2` |
| `ESLint`                               | `9.11.1` |
| `node`                                 | `20.17.0` |
